### PR TITLE
add set of character 

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
add set of character for mysql dump and binlogsyncer

### CODE REVIEW

File: canal/config.go [hunk 1]
```diff
@@ -65,6 +66,7 @@ func NewDefaultConfig() *Config {
 	c.User = "root"
 	c.Password = ""
 
+	c.Charset = "utf-8"
```
Comment (siddontang) [inline canal/config.go]: should use DEFAULT_CHARSET here?

File: canal/canal.go [hunk 2]
```diff
@@ -266,10 +273,15 @@ func (c *Canal) prepareSyncer() error {
 		Port:     uint16(port),
 		User:     c.cfg.User,
 		Password: c.cfg.Password,
+		Charset : c.cfg.Charset,
 	}
 
 	c.syncer = replication.NewBinlogSyncer(&cfg)
 
+        //if len(c.cfg.Charset) != 0 {
```
Comment (siddontang) [inline canal/canal.go]: we can remove these.

**Comment by siddontang** [comment]
Thanks @WangXiangUSTC 

Can any test help us to cover other charsets?

File: canal/canal.go (hunk 2 above)
Comment (WangXiangUSTC) [inline canal/canal.go]: yeah, I have already set character at the replication/binlogsyncer.go, these code is no need

Comment (siddontang) [inline canal/canal.go]: please remove these comments. :-)

**Comment by siddontang** [comment]
CI failed @WangXiangUSTC

**Comment by WangXiangUSTC** [comment]
It is because set charcter "utf-8" in the dump_test.go, and mysqldump can't recognize it. CI already run success after I set "utf8".

File: canal/canal.go [hunk 3]
```diff
@@ -101,6 +101,13 @@ func (c *Canal) prepareDumper() error {
 		c.dumper.AddTables(tableDB, tables...)
 	}
 
+        charset := c.cfg.Charset
+        if len(charset) == 0 {
```
Comment (siddontang) [inline canal/canal.go]: maybe `c.dumper.SetCharset(charset)` is enough here?

Comment (WangXiangUSTC) [inline canal/canal.go]: ok, I will modify this

**Comment by siddontang** [comment]
Thanks @WangXiangUSTC

**Pull request merged**